kernal: Creating a interface for getting fix lists (ifixes, tfixes) from featureManager class

jakub-pomykala opened the pull request:
- Created a new interface (`FixManager`) for getting the lists of `iFixes` and/or `tFixes`
- Adding a `atomicBoolean` to `populateFixSets()` to check if it should run or not

jakub-pomykala committed:
Creating a getter for fix lists

- Created a new interface (FixManager) for getting the lists of iFixes and/or tFixes
- Adding a atomicBoolean to check to populateFixSets() to check if it should run or not

tjwatson commented on dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/FixManager.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Set;
+
+/**
+ * <p>The IFixManager is a service for getting a set of iFixes installed in the server.</p>
```
Typo in javadoc calling it `IFixManager`

tjwatson commented on dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/FixManager.java:
```diff
@@ -0,0 +1,35 @@
+
+    /**
+     * <p>Requests the set of iFixes that are installed on the server. The result reported
+     * to the caller is never null, but may be an empty set if no features are installed.</p>
```
Should this say it may be empty if no iFixes are installed?

tjwatson commented on dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/FixManager.java:
```diff
@@ -0,0 +1,35 @@
+
+    /**
+     * <p>Requests the set of tFixes that are installed on the server. The result reported
+     * to the caller is never null, but may be an empty set if no features are installed.</p>
```
Should this say it may be empty if no tFixes are installed?

tjwatson reviewed (dismissed):
Only javadoc comments.  Not critical since we do not produce javadoc for non API/SPI classes.

jakub-pomykala commented:
#build

jakub-pomykala committed:
Fixing mistakes in the Javadocs

jakub-pomykala committed:
Adding tests for getIFixes() and getTfixes()

jakub-pomykala commented:
#build

tjwatson reviewed (dismissed):
LGTM, thanks for the test.

jakub-pomykala committed:
Adding test for multiple Test fixes in bundle

- Checking the contents of the APARs in the tests
- Changing the regex populating the list of fixes to remove the whitespaces which were previously braking
- Adding in new bundle for testing multiple testAPARs in a single bundle

jakub-pomykala commented:
#libby

jakub-pomykala commented:
Build had errors which were irrelevant to these changes.
